(Feature) GoodMarket - Add link from wallet

serdiukov-o-nordwhale opened the pull request:
# Description

- removed old market stuff
- added is first clicked flag to the user properties
- created useGoodMarket hook containing the main logic
- created components for the button and popup
- implemented button logic
- connected to analytics
- added customization props to ExplanationDialog
- added styles to the dialog & button
- fixed getFormattedDateTime() to display current date

About #2530 

# Web
 <img align="left" width="380" alt="1" src="https://user-images.githubusercontent.com/59056508/95778702-077e8e80-0c9f-11eb-8966-626f09a6e733.png" >
 <img width="380" alt="1" src="https://user-images.githubusercontent.com/59056508/95779087-c89d0880-0c9f-11eb-84be-0518ac208441.png" >

<br/>

# iOS
 <img align="left" width="380" alt="1" src="https://user-images.githubusercontent.com/59056508/95778743-1cf3b880-0c9f-11eb-89f9-97945d4c5313.png" >
 <img width="380" alt="1" src="https://user-images.githubusercontent.com/59056508/95778751-1feea900-0c9f-11eb-84ff-47ddfa8db011.png" >

# Android
 <img align="left" width="380" alt="1" src="https://user-images.githubusercontent.com/59056508/95778829-490f3980-0c9f-11eb-91bf-b08ae6f5bab5.jpg" >
 <img  width="380" alt="1" src="https://user-images.githubusercontent.com/59056508/95778821-46acdf80-0c9f-11eb-9317-3d7f28017561.jpg" >

# Animation
![ezgif-7-fe61f2396bed](https://user-images.githubusercontent.com/62208067/95851259-a6b48c00-0d5a-11eb-994d-272478816d68.gif)

About #2530 


# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [x] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

serdiukov-o-nordwhale commented:
TODO:
1. test logic
2. implement design (will delegate this to @MathiasMoralesDixtra )

serdiukov-o-nordwhale commented on src/components/common/dialogs/ExplanationDialog.js:
```diff
@@ -113,12 +114,12 @@ const mapStylesToProps = () => ({
     width: '100%',
     marginBottom: getDesignRelativeHeight(theme.sizes.defaultDouble, false),
   },
-  label: {
```
add some default label styles. as we have for a timestamp in this popup or in claim/sent/received feed popups

serdiukov-o-nordwhale commented on src/components/common/dialogs/ExplanationDialog.js:
```diff
@@ -61,15 +62,15 @@ const ExplanationDialog = ({
   const imageProps = {
     style: [
       styles.image,
-      { height: getDesignRelativeHeight(imageHeight, false) },
```
should be reverted as could break other dialogs UI

serdiukov-o-nordwhale commented on src/components/common/dialogs/ExplanationDialog.js:
```diff
@@ -53,6 +53,7 @@ const ExplanationDialog = ({
   image: ImageComponent,
   imageHeight = 74,
   buttons,
+  fullHeight = false,
```
remove this. add `containerStyle` prop instead

serdiukov-o-nordwhale commented on src/components/common/dialogs/ExplanationDialog.js:
```diff
@@ -61,15 +62,15 @@ const ExplanationDialog = ({
 
   return (
-    <View style={styles.container}>
+    <View style={[styles.container, fullHeight && styles.fullHeight]}>
```
add containerStyle to the styles array

serdiukov-o-nordwhale commented on src/components/common/dialogs/ExplanationDialog.js:
```diff
@@ -113,12 +114,12 @@ const mapStylesToProps = () => ({
   title: {
     marginBottom: getDesignRelativeHeight(8),
   },
+  fullHeight: {
```
move this style to GoodMarketDIalog, pass it via containerStyle prop

serdiukov-o-nordwhale commented on src/components/dashboard/GoodMarket/components/GoodMarketDialog.js:
```diff
@@ -0,0 +1,83 @@
+    justifyContent: 'center',
+    marginBottom: getDesignRelativeHeight(36),
+  },
+  fullHeight: {
```
```suggestion
  container: {
```

serdiukov-o-nordwhale commented on src/components/dashboard/GoodMarket/components/GoodMarketDialog.js:
```diff
@@ -0,0 +1,83 @@
+      titleStyle={styles.title}
+      textStyle={styles.text}
+      imageContainer={styles.imageContainer}
+      containerStyle={styles.fullHeight}
```
```suggestion
      containerStyle={styles.container}
```

MathiasMoralesDixtra commented:
The font has been adjusted.
![Screen Shot 2020-10-09 at 11 04 10 AM](https://user-images.githubusercontent.com/59056508/95592544-2b33a180-0a1f-11eb-8025-3060217dbaef.png)

serdiukov-o-nordwhale commented:
Checked - everything is ok. 
<img width="312" alt="image" src="https://user-images.githubusercontent.com/62208067/95597540-736eb600-0a57-11eb-9463-78512b3a82f4.png">
<img width="302" alt="image" src="https://user-images.githubusercontent.com/62208067/95597574-7ec1e180-0a57-11eb-88d9-d2d8661ba0d6.png">

moving to ready to review

serdiukov-o-nordwhale commented on netlify.toml:
```diff
@@ -10,7 +10,6 @@ REACT_APP_SERVER_URL = "https://good-server.herokuapp.com/"
 REACT_APP_NETWORK = "fuse"
 REACT_APP_SKIP_EMAIL_VERIFICATION = "true"
-REACT_APP_MARKET = "false"
 REACT_APP_ETORO = "false"
```
```suggestion
REACT_APP_MARKET_URL = ""
REACT_APP_ETORO = "false"
```

serdiukov-o-nordwhale commented on netlify.toml:
```diff
@@ -74,7 +73,6 @@ REACT_APP_SERVER_URL = "https://goodserver.gooddollar.org/"
 REACT_APP_NETWORK = "production"
 REACT_APP_SKIP_EMAIL_VERIFICATION = "false"
-REACT_APP_MARKET = "false"
 REACT_APP_ETORO = "false"
```
```suggestion
REACT_APP_MARKET_URL = ""
REACT_APP_ETORO = "false"
```

serdiukov-o-nordwhale commented on netlify.toml:
```diff
@@ -111,7 +109,6 @@ REACT_APP_SERVER_URL = "https://goodserver-qa.herokuapp.com/"
 REACT_APP_NETWORK = "staging"
 REACT_APP_SKIP_EMAIL_VERIFICATION = "false"
-REACT_APP_MARKET = "false"
 REACT_APP_ETORO = "false"
```
```suggestion
REACT_APP_MARKET_URL = ""
REACT_APP_ETORO = "false"
```

serdiukov-o-nordwhale commented on src/components/dashboard/GoodMarket/components/GoodMarketDialog.js:
```diff
@@ -0,0 +1,84 @@
+
+import ExplanationDialog from '../../../common/dialogs/ExplanationDialog'
+
+// import normalizeText from '../../../../lib/utils/normalizeText'
```
```suggestion
```

serdiukov-o-nordwhale commented on src/components/dashboard/GoodMarket/hooks/useGoodMarket.js:
```diff
@@ -0,0 +1,30 @@
+
+export default () => {
+  const { userProperties } = userStorage
+  const goToMarket = useCallback(() => openLink(marketUrl), [wasClicked])
```
```suggestion
  const goToMarket = useCallback(() => openLink(marketUrl), [])
```

sirpy commented on src/assets/GoodMarket.svg:
```diff
@@ -0,0 +1,17 @@
+<svg xmlns="http://www.w3.org/2000/svg"
```
we should already have it as icon in our fonts

sirpy commented on .env.dev:
```diff
@@ -18,7 +18,5 @@ REACT_APP_THROW_SAVE_PROFILE_ERRORS=true
 REACT_APP_WEB3_SITE_URL=https://w3.gooddollar.org
 REACT_APP_WEB3_SITE_URL_ECONOMY_ENDPOINT=/learn/economy
 REACT_APP_MNEMONIC_TO_SEED=true
-REACT_APP_MARKET_URL=https://etoro.paperclip.co
```
keep this env
with default https://www.facebook.com/groups/gooddollarmarketplace
change this to default every where also in config.js

LiavGut reviewed (changes requested):
Button on Feed: 
1. The shadow is missing  (as per design)

Popup: 
1. Title should be in the same font and weight as the other popup titles.
2. The icon is a bit too high (please check the design)
3. On web, the icon looks a bit too small.

LiavGut commented:
@sirpy Can we make the button appear by sliding up for the bottom?

sirpy commented:
> @sirpy Can we make the button appear by sliding up for the bottom?

if anything it should disappear when user scrolls. and appear when at rest or at the scrolltop
how will the user know he needs to slide up?also not natural on desktop

LiavGut commented:
> > @sirpy Can we make the button appear by sliding up for the bottom?
> 
> if anything it should disappear when user scrolls. and appear when at rest or at the scrolltop
> how will the user know he needs to slide up?also not natural on desktop

@sirpy 
Nop, didn't mean for the user to slide it. Just it's first appearance on the screen should be slide from the bottom (like basic animation). 
The request was that it will stay sticky on the dashboard, which means even when scrolling the feed.

serdiukov-o-nordwhale commented:
@MathiasMoralesDixtra i've updated PR with two adjustments requested by @sirpy 
a. re-used market icon. I created GoodMarketImage component containing an arrow image i cut off the .xd and the Icon component using 'goodmarket' icon. To remove 'GoodMarket' text from the icon i used overflow: hidden and also used margin-left to move the icon to the left, closer to the arrow. Probably you know the better way to do that - you're free to adjust this
b. updated env

Please:
1. Finish icon adjustments. Fix arrow + icon's look to be like in xd. Make sure that truncated 'GoodMarket' text doesn't appears in any screen sizes
2. Fix everything @LiavGut requested https://github.com/GoodDollar/GoodDAPP/pull/2540#pullrequestreview-506153801
3. Add basic animation. On mount GoodMarketButton should slide from the bottom. Then it should stay sticky
4. Update snapshots, make sure tests are passing

The changes above are ASAP. The following could wait once most critical goodnative bugs will be resolved:
5. Update native PR. Use svgs as the components in native implementation. Re-test the issue and make sure everything is working like in the master

serdiukov-o-nordwhale commented on src/components/dashboard/GoodMarket/components/GoodMarketButton.js:
```diff
@@ -0,0 +1,76 @@
+const GoodMarketButton = ({ styles }) => {
+  const [showDialog] = useDialog()
+  const { wasClicked, trackClicked, goToMarket } = useGoodMarket()
+  const slideAnim = useRef(new Animated.Value(-100)).current
```
```suggestion
```

serdiukov-o-nordwhale commented on src/components/dashboard/GoodMarket/components/GoodMarketButton.js:
```diff
@@ -0,0 +1,76 @@
+  }, [goToMarket])
+
+  useEffect(() => {
+    Animated.timing(slideAnim, {
```
```suggestion
    const slideAnim = new Animated.Value(-100)

    Animated.timing(slideAnim, {
```

serdiukov-o-nordwhale commented on src/components/dashboard/GoodMarket/components/GoodMarketButton.js:
```diff
@@ -0,0 +1,76 @@
+      toValue: 0,
+      duration: 3000,
+    }).start()
+  }, [slideAnim])
```
```suggestion
  }, []) // just on mount
```

serdiukov-o-nordwhale commented on src/components/dashboard/GoodMarket/components/GoodMarketButton.js:
```diff
@@ -0,0 +1,76 @@
+    marginHorizontal: 'auto',
+    borderTopRightRadius: 22,
+    borderTopLeftRadius: 22,
+    boxShadow: '-3px -3px 9px #00000029',
```
do we have that color defined in the theme (probably as rgba(0, 0, 0, 0.16) )? if yes, should use from theme.colors

serdiukov-o-nordwhale replied:
i checked - we have no var for that, rgba(0, 0, 0, 0.16)  or #00000029 is used in other places, so i'm resolving this notice

serdiukov-o-nordwhale reviewed (commented):
requesting changes

serdiukov-o-nordwhale commented:
@MathiasMoralesDixtra please update screenshots & add gif with animation shown
Thanks

serdiukov-o-nordwhale commented:
@LiavGut 

Box shadow was added:
<img width="1357" alt="image" src="https://user-images.githubusercontent.com/62208067/95841144-b1b4ef80-0d4d-11eb-8b21-c6f4c9f9a18d.png">

Font & weight adjusted to other popup titles (Roboto Slab, 700):
<img width="1210" alt="image" src="https://user-images.githubusercontent.com/62208067/95841846-81218580-0d4e-11eb-9c1a-410e7163df57.png">
<img width="1229" alt="image" src="https://user-images.githubusercontent.com/62208067/95842017-b75f0500-0d4e-11eb-8703-65872a655f83.png">

Icon height adjusted to 76px like in xd:
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/62208067/95843895-eaa29380-0d50-11eb-85aa-2c0e0e390b46.png">
<img width="952" alt="image" src="https://user-images.githubusercontent.com/62208067/95843470-69e39780-0d50-11eb-9316-6a1fb358d5fe.png">

As we're using iconic font for GoodMarket image it's unable to make it aligned exactly the same way as in design. We choosed the nearest font-size, but we anyway having a small gap (displayed by the arrow)

serdiukov-o-nordwhale commented:
Merging, as

1. tests passed
2. @LiavGut has approved the latest animations in the Slack:
<img width="506" alt="image" src="https://user-images.githubusercontent.com/62208067/95860391-ade29680-0d68-11eb-96a5-d70ba08e6498.png">
3. we have to start testing on QA just now to release it tomorrow
